### PR TITLE
[clang-format] Enable again some operator tests

### PR DESCRIPTION
**Pull request by rayroudc**
Multiple formatting operator tests are commented. This change enables them again.

[PR506629](https://llvm.org/PR50629) fixed by [D153798](https://reviews.llvm.org/D153798).

Tests in `ConfigurableSpaceBeforeParens` have the same behavior as before [D110833](https://reviews.llvm.org/D110833).

Update the test for function declaration and definition, as changed in [D114696](https://reviews.llvm.org/D114696).

### CODE REVIEW

File: clang/unittests/Format/FormatTest.cpp [hunk 1]
```diff
@@ -16671,9 +16667,8 @@ TEST_F(FormatTest, ConfigurableSpaceBeforeParens) {
                SpaceFuncDecl);
   verifyFormat("int f () throw(Deprecated);", SpaceFuncDecl);
   verifyFormat("typedef void (*cb)(int);", SpaceFuncDecl);
-  // FIXME these tests regressed behaviour.
-  // verifyFormat("T A::operator() ();", SpaceFuncDecl);
-  // verifyFormat("X A::operator++ (T);", SpaceFuncDecl);
+  verifyFormat("T A::operator()();", SpaceFuncDecl);
+  verifyFormat("X A::operator++(T);", SpaceFuncDecl);
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:16671]: Why are the spaces removed?

File: clang/unittests/Format/FormatTest.cpp [hunk 2]
```diff
@@ -16710,7 +16705,7 @@ TEST_F(FormatTest, ConfigurableSpaceBeforeParens) {
   verifyFormat("typedef void (*cb)(int);", SpaceFuncDef);
   verifyFormat("T A::operator()();", SpaceFuncDef);
   verifyFormat("X A::operator++(T);", SpaceFuncDef);
-  // verifyFormat("T A::operator() () {}", SpaceFuncDef);
+  verifyFormat("T A::operator()() {}", SpaceFuncDef);
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:16708]: Why is the space removed?

File: clang/unittests/Format/FormatTest.cpp [hunk 3]
```diff
@@ -16797,7 +16792,7 @@ TEST_F(FormatTest, ConfigurableSpaceBeforeParens) {
   verifyFormat("int f() throw (Deprecated);", SomeSpace2);
   verifyFormat("typedef void (*cb) (int);", SomeSpace2);
   verifyFormat("T A::operator()();", SomeSpace2);
-  // verifyFormat("X A::operator++ (T);", SomeSpace2);
+  verifyFormat("X A::operator++ (T);", SomeSpace2);
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:16795]: Shouldn't the space be removed here?

Comment (owenca) [reply clang/unittests/Format/FormatTest.cpp:16795]: Nvm.

File: clang/unittests/Format/FormatTest.cpp (hunk 1 above)
Comment (rayroudc) [reply clang/unittests/Format/FormatTest.cpp:16671]: [D114696](https://reviews.llvm.org/D114696) reverted the behavior for function declaration and definition, to have no space before the parentheses.

Comment (owenca) [reply clang/unittests/Format/FormatTest.cpp:16671]: It seems that D114696 restored the space in `operator` function declarations and definitions.

**Review by owenca — APPROVED** [review]

**Comment by rayroudc** [comment]
@owenca, thanks for the review!
As I do not have write access, would it be possible for you to merge this change ?

**Pull request merged**